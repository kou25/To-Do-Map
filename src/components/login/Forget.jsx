import React, { Component } from 'react'



 class Forget extends Component {
    state={
            password:'',
            password2:''
        }
        
        handleChange=(e)=>{
            this.setState({
                [e.target.id]: e.target.value
            })
        }
        handleSubmit=(e)=>{
            const { password, password2 } = this.state;
            e.preventDefault()
            console.log(this.state)
            if(password !== password2){
                alert('Password does not match !')
            }
            else{
            this.props.history.push('/');
            }
    }

    render() {
        return (
            <div className="container">
                <div data-testid="heading" className="d-sm-flex align-items-center justify-content-between mb-4">
                    <h1 style={{color:"#8B4513",  fontFamily:"cursive", margin:"30px 30%"}}>Sign In<i className="fa fa-sign-in"  style={{marginLeft:"20px", color:"#8B4513"}}></i></h1>
                </div>
                <form onSubmit={this.handleSubmit} style={{width:"60%",fontFamily:"cursive", marginLeft:"10%"}}>
                <div className="form-group">
                    <label htmlFor="exampleInputPassword1">New Password</label>
                    <input type="password" 
                    className="form-control" 
                     placeholder="Password"
                     id="password"
                     required
                     onChange={this.handleChange}/>
                </div>
                <div className="form-group">
                    <label htmlFor="exampleInputPassword">Confirm Password</label>
                    <input type="password" 
                    className="form-control" 
                     placeholder="confirm"
                     id="password2"
                     required
                     onChange={this.handleChange}/>
                </div>
                <button data-testid="button" type="submit" className="btn btn-warning mr-2 shadow"  
                >Reset</button>
                </form>
            </div>
        )
    }
}


export default Forget;