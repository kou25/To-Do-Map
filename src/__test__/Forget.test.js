import React from 'react';
import ReactDom from 'react-dom'
import Forget from '../components/login/Forget'
import {render, cleanup} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'

afterEach(cleanup);
it("render without crashing",()=>{
  const div= document.createElement("div");
  ReactDom.render(
  <Forget/>,div);
})



  it("renders button correctly", ()=>{
    const{getByTestId}=render(
    <Forget/>)
    expect(getByTestId('button')).toHaveTextContent('Reset')
  })

  it('matches snapshot', ()=>{
    const tree=renderer.create(
    <Forget/>).toJSON()
    expect(tree).toMatchSnapshot()
  })