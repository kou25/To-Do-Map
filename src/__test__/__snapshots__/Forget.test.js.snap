// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="base-container"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "marginTop": "100px",
      }
    }
  >
    <div
      className="card shadow"
      style={
        Object {
          "borderRadius": "20px",
          "height": "35rem",
          "width": "30rem",
        }
      }
    >
      <div
        className="content"
      >
        <div
          className="image"
          style={
            Object {
              "marginLeft": "25%",
              "marginTop": "40px",
            }
          }
        >
          <img
            alt="logo"
            src="forgot.png"
          />
        </div>
        <form
          onSubmit={[Function]}
          style={
            Object {
              "fontFamily": "cursive",
              "marginLeft": "10%",
              "width": "60%",
            }
          }
        >
          <div
            className="form-group mt-3 mb-4"
          >
            <label
              htmlFor="exampleInputPassword1"
            >
              Password
            </label>
            <input
              className="form-control"
              id="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
            />
          </div>
          <div
            className="form-group mt-2 "
          >
            <label
              htmlFor="exampleInputPassword"
            >
               Confirm Password
            </label>
            <input
              className="form-control"
              id="password2"
              onChange={[Function]}
              placeholder="confirm"
              required={true}
              type="password"
            />
            <button
              className="btn btn-danger shadow mt-4"
              data-testid="button"
              type="submit"
            >
              Reset 
            </button>
          </div>
        </form>
      </div>
      <div
        className="right-side"
      >
        <div
          className="inner-container"
        >
          <div
            className="text"
            style={
              Object {
                "marginLeft": "86%",
              }
            }
          >
            Forget Passwrd
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
